src/classes/TG_ValidateData_TDTM.cls

```diff
@@ -0,0 +1,306 @@
+    
+        Map<Id, List<Term_Grade__c>> courseConnIdTermGradeMap = createCourseConnTermGradeMap(termGrades);      
+        for (Course_Enrollment__c eachEnrollment: [SELECT id, contact__c, course_offering__c FROM course_enrollment__c WHERE Id IN:courseConnIdTermGradeMap.keySet()]) {       
+            for (Term_Grade__c tg : courseConnIdTermGradeMap.get(eachEnrollment.id)) {          
+                if (tg.Contact__c == eachEnrollment.Contact__c && eachEnrollment.course_offering__c != NULL) {                
+                    tg.Course_Offering__c = eachEnrollment.course_offering__c;                               
+                } else {               
+                    tg.addError(Label.NoCourseOffering);              
+                }           
+            }        
+        }    
+    }
+    
+        
+   /*******************************************************************************************************
+    * @description Queries for Contact and Course Offering, if found updates the Term Grades.
+    * If the values are not found, then throws an error.
+    * @param termGrades A List of Term Grades with no Contact and Course Offering
+    ********************************************************************************************************/
+    private void findContactCourseOffering(List<Term_Grade__c> termGrades) {
+    
+        Map<Id, List<Term_Grade__c>> courseConnIdTermGradeMap = createCourseConnTermGradeMap(termGrades);
+        for (Course_Enrollment__c eachEnrollment: [SELECT id, contact__c, course_offering__c FROM course_enrollment__c WHERE Id IN:courseConnIdTermGradeMap.keySet()]) {                   
+            for (Term_Grade__c tg : courseConnIdTermGradeMap.get(eachEnrollment.id )) {            
+                if (eachEnrollment.contact__c != NULL && eachEnrollment.course_offering__c != NULL) {              
+                    tg.contact__c = eachEnrollment.contact__c;
+                    tg.course_offering__c = eachEnrollment.course_offering__c;                   
+                } else {                   
+                    tg.addError(Label.OnlyCourseConnectionPopulated);                    
+                }
+            }    
+        }    
+    }
+    
+   
+    /*******************************************************************************************************
+    * @description Returns a map with Course Connection and List of Term Grades as Key Value pairs.
+    * @param termGrades A List of Term Grades to create a Map from.
+    * @return createCourseConnTermGradeMap
+    ********************************************************************************************************/    
+    private Map<Id, List<Term_Grade__c>> createCourseConnTermGradeMap(List<Term_Grade__c> termGrades) {
+    
+        Map<Id, List<Term_Grade__c>> createCourseConnTermGradeMap = new Map<Id, List<Term_Grade__c>>();   
+        for (Term_Grade__c each: termGrades) {       
```

Comment:
Our for loops should have some spacing around the ":" -- looks like this was probably just a typo as we're doing so in other places.